spec/factories/users.rb

```diff
@@ -2,21 +2,36 @@
 #
 # Table name: users
 #
-#  id         :uuid             not null, primary key
-#  email      :string           not null
-#  first_name :string           not null
-#  last_name  :string           not null
-#  created_at :datetime         not null
-#  updated_at :datetime         not null
+#  id                   :uuid             not null, primary key
+#  email                :string           not null
+#  first_name           :string           not null
+#  last_name            :string           not null
+#  service              :enum             default("no_service"), not null
+#  support_user_enabled :boolean          default(FALSE)
+#  created_at           :datetime         not null
+#  updated_at           :datetime         not null
 #
 # Indexes
 #
-#  index_users_on_email  (email) UNIQUE
+#  index_users_on_email_and_service  (email,service) UNIQUE
 #
 FactoryBot.define do
   factory :user do
     sequence(:email) { |n| "user#{n}@example.com" }
     first_name { "User" }
+    service { "no_service" }
     sequence(:last_name)
+
+    trait :placements do
+      service { "placements" }
+    end
+
+    trait :claims do
+      service { "claims" }
+    end
+
+    trait :support do
+      support_user_enabled { true }
+    end
   end
```

Comment:
Thinking out loud here – not a strong opinion:

I wonder if it'd be syntactically nicer to have `:claims_user` and `:placements_user` factories? 🤔

```ruby
FactoryBot.define do
  factory :user do
    sequence(:email) { |n| "user#{n}@example.com" }

    trait :support do
      support_user_enabled { true }
    end
  end

  factory :claims_user, class: "Claims::User", parent: :user do
    service { "claims" }
  end

  factory :placements_user, class: "Placements::User", parent: :user do
    service { "placements" }
  end
end
```

Then in our tests, we could:

```ruby
normal_user = build(:placements_user)
support_user = build(:placements_user, :support)
```

This feels potentially more reflective of the namespaced models that we have, because most of the time in practice I think we'll be dealing with `Placements::User` objects. This approach would result in our tests also using namespaced User objects.

Reply:
Sounds good -- just one thing. Given the comment above about not having a `no_service`, what is a normal user?

Reply:
The User factory here would be abstract. It would be invalid by itself, given the enum requirement.

Reply:
I think it is useful to be able to create a valid user without specifying the service if the service doesn't matter. Like the parent `User` model specs. I've added `service { %w[placements claims].sample }` so that it's not an abstract.